### ISSUE TITLE
Integration test failures in master.

### ISSUE THREAD

**Issue by ph** [elastic/logstash#6959]
Proposal add a retry mechanism around the code.

```
Failures:
  1) Test Elasticsearch output can ingest 37K log lines of sample apache logs
     Failure/Error: expect(result["hits"]["total"]).to eq(37)
       expected: 37
            got: 0
       (compared using ==)
     # ./specs/es_output_how_spec.rb:25:in `(root)'
     # /home/travis/.rvm/gems/jruby-1.7.25/gems/rspec-wait-0.0.9/lib/rspec/wait.rb:46:in `(root)'
Finished in 11 minutes 32 seconds (files took 6.51 seconds to load)
34 examples, 1 failure
Failed examples:
rspec ./specs/es_output_how_spec.rb:17 # Test Elasticsearch output can ingest 37K log lines of sample apache logs
Randomized with seed 47440

**Comment by ph** [comment]
Got a local repro case 💃

**Comment by ph** [comment]
OK, found the problem here, I didn't encounter this error before because my tar for the integration suite was bit older.

```
# AUTOGENERATED BY THE GRADLE SCRIPT. DO NOT EDIT.

require 'jar_dependencies'
require_jar('org.logstash.filters', 'logstash-filter-date', '3.1.0')
```

**Comment by ph** [comment]
This the previous content was from `/vendor/bundle/jruby/1.9/gems/logstash-filter-date-3.1.4`. note the plugin version is 3.1.3.

So when we were trying to use the date filter it was throwing this error:

```
46:[2017-04-26T13:02:43,262][ERROR][logstash.plugins.registry] Problems loading a plugin with {:type=>"filter", :name=>"date", :path=>"logstash/filters/date", :error_message=>"\n\n\tyou might need to reinstall the gem which depends on the missing jar or in case there is Jars.lock then resolve the jars with `lock_jars` command\n\nno such file to load -- org/logstash/filters/logstash-filter-date/3.1.0/logstash-filter-date-3.1.0 (LoadError)", :error_class=>RuntimeError, :error_backtrace=>["/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.3.11/lib/jar_dependencies.rb:348:in `do_require'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.3.11/lib/jar_dependencies.rb:255:in `require_jar'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.3.11/lib/jar_dependencies.rb:0:in `require_jar_with_block'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/jar-dependencies-0.3.11/lib/jar_dependencies.rb:254:in `require_jar'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/lib/bootstrap/patches/jar_dependencies.rb:6:in `require_jar'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/logstash-filter-date-3.1.4/lib/logstash-filter-date_jars.rb:4:in `(root)'", "org/jruby/RubyKernel.java:1040:in `require'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/logstash-filter-date-3.1.4/lib/logstash/filters/date.rb:1:in `(root)'", "org/jruby/RubyKernel.java:1040:in `require'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/polyglot-0.3.5/lib/polyglot.rb:65:in `require'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/vendor/bundle/jruby/1.9/gems/logstash-filter-date-3.1.4/lib/logstash/filters/date.rb:5:in `(root)'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/plugins/registry.rb:1:in `(root)'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/plugins/registry.rb:151:in `legacy_lookup'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/plugins/registry.rb:133:in `lookup'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/plugins/registry.rb:175:in `lookup_pipeline_plugin'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/plugin.rb:139:in `lookup'", "org/jruby/RubyKernel.java:1079:in `eval'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline.rb:101:in `plugin'", "(eval):16:in `initialize'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline.rb:69:in `initialize'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline.rb:163:in `initialize'", "org/jruby/RubyArray.java:1613:in `each'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/pipeline_action/create.rb:27:in `execute'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/agent.rb:301:in `converge_state'", "org/jruby/ext/thread/Mutex.java:149:in `synchronize'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/agent.rb:286:in `converge_state'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/agent.rb:150:in `converge_state_and_update'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/agent.rb:148:in `converge_state_and_update'", "org/jruby/RubyProc.java:281:in `call'", "/Users/ph/es/logstash/build/logstash-6.0.0-alpha1-SNAPSHOT/logstash-core/lib/logstash/agent.rb:86:in `execute'"]}
47:[2017-04-26T13:02:43,274][ERROR][logstash.agent           ] Failed to execute action {:action=>LogStash::PipelineAction::Create/pipeline_id:main, :exception=>"LogStash::PluginLoadingError", :message=>"Couldn't find any filter plugin named 'date'. Are you sure this is correct? Trying to load the date filter plugin resulted in this error: Problems loading the requested plugin named date of type filter. Error: RuntimeError \n\n\tyou might need to reinstall the gem which depends on the missing jar or in case there is Jars.lock then resolve the jars with `lock_jars` command\n\nno such file to load -- org/logstash/filters/logstash-filter-date/3.1.0/logstash-filter-date-3.1.0 (LoadError)"}
```

**Comment by ph** [comment]
OK, date filter 3.1.4 is a broken release and doesn't include the jars.

**Comment by ph** [comment]
I've checked the last few releases and none of them include the date filter 3.1.4, I will yank the gem.

**Comment by ph** [comment]
I will close this issue since it will be resolved with the publish of the date filter 3.1.5
